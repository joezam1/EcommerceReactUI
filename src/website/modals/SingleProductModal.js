import React,{useEffect, useState} from 'react';
import DataStoreInvoker from '../../dataStore/DataStoreInvoker.js';
import Utils from '../../library/Utils.js';
import ReactRenderHelper from '../../library/ReactRenderHelper.js';


export default function SingleProductModal(){
    
    const abortController = new AbortController();
    const [item, setSingleProductAsync] = useState();

    var mounted = false;
    function cleanup(){
        mounted = false;
        abortController.abort();
    }
    

    useEffect(function(){
       mounted = true;

        var selectedProduct = DataStoreInvoker.getDataStoreState('singleProduct');
        
        setSingleProductAsync(selectedProduct);
             
       return cleanup();
    },[]);
   
    function closeSingleProductWindow(){ 
        ReactRenderHelper.closeSelectedModal();
    }
    var itemIsValid = Utils.isValid(item);
    var displayProduct = (itemIsValid)? "single-product-display":"single-product-hide";

    var product = (itemIsValid)? (
            <div key={item.ProductId} className={"single-product-card " +displayProduct} id={item.ProductId}>
            <div className="single-product-buy-button">Buy Now</div>
            <div className="single-product-container" id={item.ProductId} >
                <span id={item.ProductId} className="single-product-price-item">$ {item.Price}</span>
                <img src={'./src/assets/products/' + item.ImageHref1} id={item.ProductId} className="single-product-image"/>
                <span id={item.ProductId} className="single-product-name">{item.Name}</span>
                <span id={item.ProductId} data-sku={item.SKU} className="single-product-sku"></span>
                <span id={item.ProductId} data-serial-number={item.SerialNumber}className="single-product-serial-number"></span>
                <span id={item.ProductId} className="single-product-description">{item.Description}</span>
                <div className="back-to-store-button" onClick={closeSingleProductWindow}> Back To Store</div>

            </div>
            </div>
    ) : "";
    return(<div className="single-product-modal">
        {product}
        </div>)
}