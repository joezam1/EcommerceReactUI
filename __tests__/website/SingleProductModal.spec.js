import React from 'react';
import {render, cleanup, fireEvent, waitFor} from '@testing-library/react';
import SingleProductModal from '../../src/website/modals/SingleProductModal.js';

xdescribe('File SingleProductModal', function(){
    test('take a snapshot', function(){
        let container = render(<SingleProductModal/>);
        expect(container.asFragment(<SingleProductModal/>)).toMatchSnapshot();
    })
})