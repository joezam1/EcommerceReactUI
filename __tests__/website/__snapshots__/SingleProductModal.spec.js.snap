// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`File SingleProductModal take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="single-product-modal"
  >
    <div
      class="single-product-card single-product-display"
    >
      <div
        class="single-product-buy-button"
      >
        Buy Now
      </div>
      <div
        class="single-product-container"
      >
        <span
          class="single-product-price-item"
        >
          $ 
        </span>
        <img
          class="single-product-image"
          src="./assets/products/undefined"
        />
        <span
          class="single-product-name"
        />
        <span
          class="single-product-sku"
        />
        <span
          class="single-product-serial-number"
        />
        <span
          class="single-product-description"
        />
        <div
          class="back-to-store-button"
        >
           Back To Store
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
